arches/app/models/resource.py

```diff
@@ -193,24 +193,38 @@ def bulk_save(resources, transaction_id=None):
             resource.tiles = resource.get_flattened_tiles()
             tiles.extend(resource.tiles)
 
-        # need to save the models first before getting the documents for index
+        # need to handle if the bulk load is appending tiles to existing resources/
+        existing_resources = Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources])
```

Comment:
@aj-he I would suggest using a "values_list" here: https://docs.djangoproject.com/en/3.2/ref/models/querysets/#values-list
to get a list of existing_resource_ids directly.
Something like this:

```
existing_resources_ids = Resource.objects.filter(resourceinstanceid__in=[resource.resourceinstanceid for resource in resources]).values_list('resourceinstanceid', flat=True)
```

Which should be able to replace lines 197 and 198 and reduce the overhead of creating model instances

Reply:
@apeters thanks, I've used the `value_list` function to bypass returning resources from the DB and can now split the in mem resources list between new and existing.